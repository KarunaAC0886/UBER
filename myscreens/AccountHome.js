import { View, Text, StatusBar, Image, TouchableOpacity } from 'react-native'
import React from 'react'
import styles from './styles'
import  {FontAwesome}  from '@expo/vector-icons';
import { useNavigation } from '@react-navigation/native';


const AccountHome = () => {
    const navigation = useNavigation();
    return (
        <View style={styles.acnt_main_cntr}>
            <View style={styles.acnt_frst_cont}>
                <View style={styles.acnt_prfl_cont}>
                    <TouchableOpacity style={styles.acnt_prfl_subcont1}>
                        <Image style={styles.acnt_prfl_img} source={require('../assets/profile.png')} />
                    </TouchableOpacity>
                    <View style={styles.acnt_prfl_subcont2}>
                        <Text style={styles.acnt_prfl_nm}>Krishna</Text>
                        <Text style={styles.acnt_prfl_nm}>+91 8123456789</Text>
                    </View>
                </View>
                <View style={styles.acnt_frst_msg_cont}>
                <TouchableOpacity onPress={()=>{
                    navigation.navigate('messages');
                }} style={styles.acnt_frst_msg_btn}>
                <View><Text style={styles.acnt_frst_msglbl}>Messages</Text></View><FontAwesome style={styles.acnt_frst_rgtarrow} name='chevron-right'/>
                </TouchableOpacity>
                </View>
            </View>
            <View style={styles.acnt_scnd_cont}>
                <View style={styles.acnt_scnd_cont_sub1}>
                <TouchableOpacity style={styles.acnt_scnd_cont_sub1_btnactns} onPress={()=>{
                    navigation.navigate('trips')
                }}><Text style={styles.acnt_scnd_cont_sub1_btntxt}>Your Trips</Text></TouchableOpacity>
                <TouchableOpacity style={styles.acnt_scnd_cont_sub1_btnactns}
                onPress={()=>{navigation.navigate('Payment')}}><Text style={styles.acnt_scnd_cont_sub1_btntxt}>Payment</Text></TouchableOpacity>
                <TouchableOpacity style={styles.acnt_scnd_cont_sub1_btnactns} onPress={()=>{
                    navigation.navigate('UberPass')
                }}><Text style={styles.acnt_scnd_cont_sub1_btntxt}>Uber Pass</Text></TouchableOpacity>
                <TouchableOpacity style={styles.acnt_scnd_cont_sub1_btnactns} onPress={()=>{
                    navigation.navigate('accountsettings')
                }}><Text style={styles.acnt_scnd_cont_sub1_btntxt}>Settings</Text></TouchableOpacity>
                </View>
                <View style={styles.acnt_scnd_cont_sub2}>
                <Text style={styles.acnt_scnd_cont_sub1_btntxt}>Legal</Text>
                <Text style={styles.acnt_scnd_cont_sub1_legaltxt}>v4.3722003</Text>
                </View>
            </View>
            <StatusBar />
        </View>
    )
}

export default AccountHome