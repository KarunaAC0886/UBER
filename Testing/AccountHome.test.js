import 'react-native';
import React from 'react';
import AccountHome from '../myscreens/AccountHome'
import renderer from 'react-test-renderer';
import { useNavigation } from '@react-navigation/native';


jest.mock('@expo/vector-icons', () => ({
    FontAwesome: 'FontAwesome', 
}));

jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
}));

jest.mock('react-native-responsive-screen', () => ({
  widthPercentageToDP: jest.fn(),
  heightPercentageToDP: jest.fn(),
}));

describe('AccountHome', () => {
  it('AccountHome page renders correctly', () => {
    useNavigation.mockReturnValue({ navigate: jest.fn() });

    const tree = renderer.create(<AccountHome/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});