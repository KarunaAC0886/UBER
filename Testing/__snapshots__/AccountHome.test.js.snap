// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountHome AccountHome page renders correctly 1`] = `
<View
  style={
    {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#000000",
        "height": "30%",
        "paddingHorizontal": "6%",
        "paddingTop": "6%",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "borderBottomColor": "#EDF6FF",
          "borderBottomWidth": 1,
          "flex": 0,
          "flexDirection": "row",
          "height": "65%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "flex": 0.5,
            "height": 100,
            "opacity": 1,
            "width": 100,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/profile.png",
            }
          }
          style={
            {
              "borderRadius": 50,
              "height": 100,
              "width": 100,
            }
          }
        />
      </View>
      <View
        style={
          {
            "flex": 1,
            "paddingTop": "7%",
            "textAlign": "left",
          }
        }
      >
        <Text
          style={
            {
              "color": "#EDF6FF",
              "fontSize": 23,
            }
          }
        >
          Krishna
        </Text>
        <Text
          style={
            {
              "color": "#EDF6FF",
              "fontSize": 23,
            }
          }
        >
          +91 8123456789
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flex": 0,
          "flexDirection": "row",
          "height": "35%",
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "flex": 0,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <View>
          <Text
            style={
              {
                "color": "#ffffff",
                "fontSize": 20,
              }
            }
          >
            Messages
          </Text>
        </View>
        <FontAwesome
          name="chevron-right"
          style={
            {
              "color": "#ffffff",
              "fontSize": 20,
              "paddingRight": 30,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#1A1A1A",
        "flex": 0,
        "flexDirection": "column",
        "height": "70%",
        "justifyContent": "space-between",
        "padding": "6%",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "flex": 0,
          "flexDirection": "column",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
            "paddingVertical": "4%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#ffffff",
              "fontSize": 20,
            }
          }
        >
          Your Trips
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
            "paddingVertical": "4%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#ffffff",
              "fontSize": 20,
            }
          }
        >
          Payment
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
            "paddingVertical": "4%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#ffffff",
              "fontSize": 20,
            }
          }
        >
          Uber Pass
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
            "paddingVertical": "4%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#ffffff",
              "fontSize": 20,
            }
          }
        >
          Settings
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flex": 0,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#ffffff",
            "fontSize": 20,
          }
        }
      >
        Legal
      </Text>
      <Text
        style={
          {
            "color": "#ffffff",
            "fontSize": 14,
          }
        }
      >
        v4.3722003
      </Text>
    </View>
  </View>
</View>
`;
